src/axolotl/utils/trainer.py

```diff
@@ -608,6 +656,8 @@ def setup_trainer(cfg, train_dataset, eval_dataset, model, tokenizer, total_num_
     trainer_cls = (
         OneCycleLRSchedulerTrainer
         if cfg.lr_scheduler == "one_cycle" and (cfg.fsdp or cfg.adapter == "qlora")
+        else ReLoRATrainer
+        if cfg.relora_steps
```

Comment:
I recommend refactor to a normal if elif instead of this long chain

Reply:
Done, and I've added a check for overlapping one_cycle + relora in `validate_config`.